### PR TITLE
10.0 fields.Binary: onchange('binary_field') get string u'1520 bytes' instead of binary base64 string

### PR DESCRIPTION
**Pull request by moylop260**
Impacted versions: 10.0

Steps to reproduce:

1. Create a fields.Binary: `my_file_bin = fields.Binary()`
2. Create other field: `name = fields.Char()`
2. Add a onchange method to `name` field and check the value of `my_file_bin` field:
```python
@api.onchange('name')
def onchange_name(self):
    print self.my_file_bin
```
3. Create a record and save
4. Edit change the name and press tab.

Current behavior:
`self.my_file_bin: u'1520 bytes'`

Expected behavior:
`self.my_file_bin: {{base64_string}}`

Video/Screenshot link (optional):
screen shot 2016-11-22 at 4 14 32 pm

[IMAGE ALT TEXT](http://www.youtube.com/watch?v=3Hau3EW6KsI "Video Title")

### CODE REVIEW

**Comment by moylop260** [comment]
Thanks for feedback.

I can reproduce it installing just the `binary_test` module, but just for records with id (exists)
I'm creating a PR for odoo/odoo to get a runbot build and record a video using directly runbot.

**Comment by moylop260** [comment]
The video using runbot.odoo.com
[IMAGE ALT TEXT](http://www.youtube.com/watch?v=SFkdXHt13xA "Video Title")

**Comment by mart-e** [comment]
Hello,

Thanks, after a bit more search, I think I understand the issue.
When you first upload the attachment, the content is loaded. The ORM can fetch the value as it just got it from the client.
Now once you have saved, the record you had in memory when creating is discarded and reloaded via a `search_read`.

As you can have very heavy attachments (gigabytes), we do not load the attachment content when loading the record, we only load its size.
Now when you trigger a onchange, you get the values of every field in the state known by the web client, so for your field `value`, you only get the size.

Long story short, in an onchange, we an guarantee you will be handling the real file content but may have only its description (size).

**Comment by moylop260** [comment]
Then this is a expected behavior and we should avoid use binary fields into a onchange method, right?
